modules/account-lib/src/coin/dot/keyPair.ts

```diff
@@ -12,18 +19,56 @@ export class KeyPair extends Ed25519KeyPair {
     super(source);
   }
 
+  /**
+   * Helper function to create the KeyringPair for signing a dot transaction.
+   *
+   * @returns {KeyringPair} dot KeyringPair
+   * @see https://polkadot.js.org/docs/api/start/keyring
+   */
+  protected createPolkadotPair(): KeyringPair {
+    const secretKey = this.keyPair.prv ? new Uint8Array(Buffer.from(this.keyPair.prv, 'hex')) : new Uint8Array();
+    const publicKey = new Uint8Array(Buffer.from(this.keyPair.pub, 'hex'));
```

Comment:
Is it possible there is no `this.keyPair.pub`?

Reply:
nope, `pub` will always be present, only `prv` is optional incase we build an account-lib keypair with just a public key